apps/prairielearn/src/pages/publicAssessments/publicAssessments.html.ts

```diff
@@ -2,18 +2,90 @@ import { html } from '@prairielearn/html';
 
 import { AssessmentModuleHeading } from '../../components/AssessmentModuleHeading.html.js';
 import { AssessmentSetHeading } from '../../components/AssessmentSetHeading.html.js';
+import { Modal } from '../../components/Modal.html.js';
 import { PageLayout } from '../../components/PageLayout.html.js';
+import type { CopyTarget } from '../../lib/copy-content.js';
 import type { CourseInstance } from '../../lib/db-types.js';
 import { type AssessmentRow } from '../../models/assessment.js';
 
+function CopyCourseInstanceModal({
+  courseInstance,
+  courseInstanceCopyTargets,
+}: {
+  courseInstance: CourseInstance;
+  courseInstanceCopyTargets: CopyTarget[] | null;
+}) {
+  if (courseInstanceCopyTargets == null) return '';
+  return Modal({
+    id: 'copyCourseInstanceModal',
+    title: 'Copy course instance',
+    formAction: courseInstanceCopyTargets[0]?.copy_url ?? '',
+    formClass: 'js-copy-question-form',
```

Comment:
I need to make sure the frontend JS here is working properly, and possibly rename the selector to `copy-content-form`.

Reply:
done in b8f81a15bb35c5acaa58a4ce9f9e26cdcd398282